sdk/python/tests/compiler/component_builder_test.py

```diff
@@ -110,6 +102,28 @@ def test_generate_requirement(self):
     golden_requirement_payload = '''\
 tensorflow >= 0.10.0, <= 0.11.0
 kubernetes >= 0.6.0
+'''
+    with open(temp_file, 'r') as f:
+      target_requirement_payload = f.read()
+    self.assertEqual(target_requirement_payload, golden_requirement_payload)
+    os.remove(temp_file)
+
+  def test_add_python_package(self):
+    """ Test add_python_package """
+
+    # prepare
+    test_data_dir = os.path.join(os.path.dirname(__file__), 'testdata')
+    temp_file = os.path.join(test_data_dir, 'test_requirements.tmp')
+
+    dependency_helper = DependencyHelper()
+    dependency_helper.add_python_package(dependency=VersionedDependency(name='tensorflow', min_version='0.10.0', max_version='0.11.0'))
+    dependency_helper.add_python_package(dependency=VersionedDependency(name='kubernetes', min_version='0.6.0'))
```

Comment:
Good point. Done

Reply:
I do not see max version only test.

Reply:
I can do that. However, I want to point out that it might make sense to cover 70-80% unit tests, then it starts to affect the productivity more than the benefits it bring.

Reply:
If I were adding tests for the Versioned dependency class, I'd just do a compact test that testes all version cases:

* no version
* min_version
* max_version
* version
* min_version + max_version
* min_version + version (what's the expected behavior?)
* max_version + version (what's the expected behavior?)
* min_version + max_version + version (what's the expected behavior?)

For `override` I'd test overriding dependency with just `min_version` with a dependency with only `max_version`.